src/lib.rs

```diff
@@ -191,15 +207,20 @@ pub async fn run(config: Config) -> Result<()> {
         let images = rt.podman.as_ref().unwrap().images();
         let mut stream = images.pull(&pull_opts);
 
-        while let Some(pull_result) = stream.next().await {
-            match pull_result {
-                Ok(output) => {
-                    debug!("🔧 {:?}", output);
-                }
-                Err(e) => {
-                    error!("❌ pulling image: {}", e);
+        let present_local = false;
+        if !present_local || config.force_pull {
+            while let Some(pull_result) = stream.next().await {
+                match pull_result {
+                    Ok(output) => {
+                        debug!("🔧 {:?}", output);
+                    }
+                    Err(e) => {
+                        return Err(anyhow!("❌ error pulling image: {}", e));
+                    }
                 }
             }
+        } else {
+            debug!("🔧 skipping the pull process");
```

Comment:
let's have the same debug line for both docker and podman in case the image was found. I like the docker one that you had with the green checkmark